tests/test_flags.py

```diff
@@ -0,0 +1,324 @@
+
+        third = ins
+        ins ^= other
+        assert ins is third
+        assert ins.value == 0b011
+
+        ins.two = False
+        other.one = True
+        ins ^= other
+        assert ins.value == 0b010
+
+    def test__le__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        assert not ins <= other
+        other.one = True
+        assert ins <= other
+
+        with pytest.raises(
+            TypeError, match="'<=' not supported between instances of 'TestFlags' and 'int'"
+        ):
+            _ = ins <= 4  # type: ignore
+
+        other.value = ins.value
+        assert ins <= other
+
+    def test__ge__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        assert not ins >= other
+        ins.two = True
+        assert ins >= other
+
+        with pytest.raises(
+            TypeError, match="'>=' not supported between instances of 'TestFlags' and 'int'"
+        ):
+            _ = ins >= 4  # type: ignore
+
+        other.value = ins.value
+        assert ins >= other
+
+    def test__lt__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        assert not ins < other
+        other.one = True
+        assert ins < other
+
+        with pytest.raises(
+            TypeError, match="'<' not supported between instances of 'TestFlags' and 'int'"
+        ):
+            _ = ins < 4  # type: ignore
+
+        other.value = ins.value
+        assert not ins < other
+
+    def test__gt__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        assert not ins > other
+        ins.two = True
+        assert ins > other
+
+        with pytest.raises(
+            TypeError, match="'>' not supported between instances of 'TestFlags' and 'int'"
+        ):
+            _ = ins > 4  # type: ignore
+
+        other.value = ins.value
+        assert not ins > other
+
+    def test__invert__(self) -> None:
+        ins = TestFlags(one=True)
+        assert ins.value == 0b0001
+        other = ~ins
+        # assert that invert does not modify anything in-place
+        assert ins.value == 0b0001
+        # the other `0` here is because invert does not invert values that are not defined
+        # assert other.value == goal
+        assert other.value == 0b10110
+
+    def test__hash__(self) -> None:
+        ins = TestFlags(one=True)
+        assert hash(ins) == hash(ins.value)
+
+    def test_iter(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        ran_at_least_once = False
+
+        assert len(list(iter(ins))) == 4
+
+        for flag, value in iter(ins):
+            ran_at_least_once = True
+            assert flag in ins.VALID_FLAGS
+            assert getattr(ins, flag) == value
+
+        assert ran_at_least_once
```

Comment:
```suggestion
        assert len(list(iter(ins))) == 4

        for flag, value in iter(ins):
            assert flag in ins.VALID_FLAGS
            assert getattr(ins, flag) == value
```

Reply:
I added the check and then proceeded to not remove the run at least once checks

Reply:
https://github.com/DisnakeDev/disnake/pull/616/commits/65afd7a159061534ba4b05ccac13c7c4b5631dbb